posts/2022-09-22-Rust-1.64.0.md

```diff
@@ -0,0 +1,312 @@
+
+These APIs are now usable in const contexts:
+
+- [`slice::from_raw_parts`](https://doc.rust-lang.org/stable/core/slice/fn.from_raw_parts.html)
+
+### Compatibility notes
+
+Rust 1.64 includes some changes that might prevent you from upgrading:
+
+* As [previously
+  announced](https://blog.rust-lang.org/2022/08/01/Increasing-glibc-kernel-requirements.html),
+  `linux` targets now require at least Linux kernel 3.2 (except for targets which
+  already required a newer kernel), and `linux-gnu` targets now require glibc
+  2.17 (except for targets which already required a newer glibc).
+
+* Rust 1.64.0 changes the memory layout of `Ipv4Addr`, `Ipv6Addr`,
+  `SocketAddrV4` and `SocketAddrV6` to be more compact and memory efficient.
+   This internal representation was never exposed, but some crates relied on
+   it anyway by using `std::mem::transmute`, resulting in invalid memory
+   accesses. Such internal implementation details of the standard library are
+   *never* considered a stable interface. To limit the damage, we worked with
+   the authors of all of the still-maintained crates doing so to release fixed
+   versions, which have been out for more than a year. The vast majority of
+   impacted users should be able to mitigate with a `cargo update`.
+
+As part of the [RLS
+deprecation](https://blog.rust-lang.org/2022/07/01/RLS-deprecation.html), this
+is also the last release containing a copy of RLS. Starting from Rust 1.65.0,
+RLS will be replaced by a small LSP server showing the deprecation warning.
+
+### Other changes
+
+There are other changes in the Rust 1.64 release, including:
+
+* Windows builds of the Rust compiler now use profile-guided optimization,
+  providing performance improvements of 10-20% for compiling Rust code on
+  Windows.
+
+* If you define a struct containing fields that are never used, rustc will warn
+  about the unused fields. Now, in Rust 1.64, you can enable the
+  `unused_tuple_struct_fields` lint to get the same warnings about unused
+  fields in a tuple struct. In future versions, we plan to make this lint
+  warn by default.  Fields of type unit (`()`) do not produce this warning,
+  to make it easier to migrate existing code without having to change tuple
+  indices.
+
+* When working with collections of related libraries or binary crates in one
+  Cargo workspace, you can now avoid duplication of common field values between
+  crates, such as common version numbers, repository URLs, or `rust-version`.
+  This also helps keep these values in sync between crates when updating them.
+  For more details, see
+  [`workspace.package`](https://doc.rust-lang.org/cargo/reference/workspaces.html#the-workspacepackage-table),
+  [`workspace.dependencies`](https://doc.rust-lang.org/cargo/reference/workspaces.html#the-workspacedependencies-table),
+  and ["inheriting a dependency from a
+  workspace"](https://doc.rust-lang.org/cargo/reference/specifying-dependencies.html#inheriting-a-dependency-from-a-workspace).
```

Comment:
I believe [this beta backport](https://github.com/rust-lang/cargo/pull/11097) changed the links around. The change isn't showing in the [beta docs](https://doc.rust-lang.org/beta/cargo/reference/workspaces.html) but they are showing in the [nightly docs](https://doc.rust-lang.org/nightly/cargo/reference/workspaces.html)
```suggestion
  [`workspace.package`](https://doc.rust-lang.org/cargo/reference/workspaces.html#the-package-table),
  [`workspace.dependencies`](https://doc.rust-lang.org/cargo/reference/workspaces.html#the-dependencies-table),
  and ["inheriting a dependency from a
  workspace"](https://doc.rust-lang.org/cargo/reference/specifying-dependencies.html#inheriting-a-dependency-from-a-workspace).
```